aot-maven-plugin/src/main/java/com/dylibso/chicory/maven/aot/AotGenMojo.java

```diff
@@ -0,0 +1,275 @@
+package com.dylibso.chicory.maven.aot;
+
+import static com.github.javaparser.StaticJavaParser.parseType;
+
+import com.dylibso.chicory.aot.AotMachine;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.wasm.Parser;
+import com.dylibso.chicory.wasm.types.ValueType;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.ArrayAccessExpr;
+import com.github.javaparser.ast.expr.ArrayCreationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.AssignExpr;
+import com.github.javaparser.ast.expr.CastExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.IntegerLiteralExpr;
+import com.github.javaparser.ast.expr.LambdaExpr;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.NullLiteralExpr;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.utils.SourceRoot;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * This plugin generates an invokable library from the compiled Wasm
+ */
+@Mojo(name = "wasm-aot-gen", defaultPhase = LifecyclePhase.GENERATE_RESOURCES, threadSafe = true)
```

Comment:
`GENERATE_SOURCES` is probably more appropriate, since we're generating sources and that phase runs earlier

Reply:
ok, agreed, thinking about this line twice, I'm not sure if this is `thread-safe`, I change it to `false` to stay on the safe side for now.

Reply:
It should be thread safe. According to [this post](https://stackoverflow.com/questions/50247168/how-to-make-sure-that-a-maven-plugin-is-thread-safe):

> a mojo is considered `threadSafe` as long as there's no mutable state that is *shared across different instances* of the mojo -- e.g. a static reference to an object with mutable contents

The expectation now for Maven is that every Mojo should be thread safe. You have to violate general Java best practices to make it not thread safe.

Reply:
got it, done, thanks for looking it up :+1: